src/main/java/com/hms_networks/americas/sc/extensions/eventfile/EventFile.java

```diff
@@ -0,0 +1,130 @@
+      throws IOException, TimeoutException, JSONException {
+    return EventDidOccurInDurationMins(
+        EventFileConstants.DEFAULT_DURATION_MINS, EventFileConstants.FILE_CIRCULARIZED_EVENT_ID);
+  }
+
+  /**
+   * Read EventFile, check for event
+   *
+   * @param eventId event id to look for
+   * @return boolean - did the event occur in the time frame
+   * @throws TimeoutException for timeout in EBD read
+   * @throws IOException for errors in parsing response
+   * @throws JSONException for error parsing JSON
+   */
+  public static boolean EventDidOccur(int eventId)
+      throws IOException, TimeoutException, JSONException {
+    return EventDidOccurInDurationMins(EventFileConstants.DEFAULT_DURATION_MINS, eventId);
+  }
+
+  /**
+   * Read EventFile, check for event
+   *
+   * @param mins duration in minuets to check for event id
+   * @param eventId event id to look for
+   * @return boolean - did the event occur in the time frame
+   * @throws TimeoutException for timeout in EBD read
+   * @throws IOException for errors in parsing response
+   * @throws JSONException for error parsing JSON
+   */
+  public static boolean EventDidOccurInDurationMins(int mins, int eventId)
+      throws IOException, TimeoutException, JSONException {
+
+    if (mins < EventFileConstants.MIN_DURATION_MINS
+        || mins > EventFileConstants.MAX_DURATION_MINS) {
+      throw new IllegalArgumentException(
+          "The parameter passed must be greater than 0, but less than 90.");
+    }
+
+    String ebdStr = "$dtEV$ftT$st_m" + Integer.toString(mins);
+    Exporter ex = HistoricalDataManager.ExecuteEBDCall(ebdStr);
+    return parseEventFileExportResponse(ex, Long.toString(eventId));
+  }
+
+  /**
+   * Parses Export Block Descriptor Event Log response, checks for event
+   *
+   * @param exporter EBD Exporter
+   * @param eventId ID of the target event as a string
+   * @return was event ID found
+   * @throws IOException for parsing exceptions
+   * @throws JSONException for JSON parsing exceptions
+   */
+  private static boolean parseEventFileExportResponse(Exporter exporter, String eventId)
+      throws IOException, JSONException {
+    final DataInputStream dataStream = new DataInputStream(exporter);
+    final BufferedReader reader = new BufferedReader(new InputStreamReader(dataStream));
+
+    // Create line counter and variable to store current line
+    int lineCount = 0;
+    String line = reader.readLine();
+    // Loop through lines in file until end and store data points
+    while (line != null) {
+
+      // Only parse lines 1 and greater, skip header
+      if (lineCount > 0) {
+        // Parse line
+        if (checkEventFileLine(line, eventId) == true) {
+          return true;
+        }
+      }
+
+      // Increment line count
+      lineCount++;
+      // Read next line before looping again
+      line = reader.readLine();
+    }
+    reader.close();
+    exporter.close();
+    return false;
```

Comment:
```suggestion
    final String exporterFile = StringUtils.getStringFromInputStream(exporter, "UTF-8");
    final List eventFileLines = StringUtils.split(exporterFile, "\n");
    for (int x = 0; x < eventFileLines.size(); x++) {
      String eventFileLine = (String) eventFileLines.get(x);
      return checkEventFileLine(eventFileLine, eventId);
    }

    exporter.close();
    return false;
```

I think it would be better to use the StringUtils function for reading the exporter. It allows for an overall simpler and smaller method.

Reply:
I care more about memory efficiency. StringUtils copies the bytes into a new string. Then copies parts of that string in to a new arrayList. This means that every byte from the stream is copied twice.  With all the Flexy memory and thread starvation uncertainty, my preference is for memory efficiency and performance. My hope is that bufferedReader is somehow more efficient with memory.  I really am not familiar with Java so I will make the change if you feel it is the better choice.

Reply:
I designed the StringUtils.getStringFromInputStream method with performance in mind. I did research online and generally noticed that ByteArrayOutputStream/InputStream.read was the most efficient out of most solutions. Additionally, BufferedReader can/will modify the contents to convert any line separators to the system line separator, which we may not want coming from a stream, where we expect consistent output.

See: https://stackoverflow.com/questions/309424/how-do-i-read-convert-an-inputstream-into-a-string-in-java

Reply:
Additionally, if there is a memory or performance concern with something in the library, I think we should focus on resolving it in the shared method instead of duplicating code across the library that does the same thing(s).

Reply:
Agreed. What is the suggestion?  Can we complete this PR, write a spec... then make another PR.

Reply:
This could be completed in another PR, but the suggestion is that we use the StringUtils method, as it has been benchmarked to perform the best. Any other implementation or change of reading a string from an input stream should be separately addressed if we find it necessary.

Reply:
I don't think we should introduce code here which performs the same functionality as code we already have and can reference. The referenced method should be addressed (if necessary), instead of creating a new one.